src/istio/control/tcp/attributes_builder.cc

```diff
@@ -32,6 +32,8 @@ void AttributesBuilder::ExtractCheckAttributes(CheckData* check_data) {
     builder.AddInt64(AttributeName::kSourcePort, source_port);
   }
 
+  // TODO(diemtvu): add TCP authn filter similar to http case, and use authn
+  // result output here instead.
```

Comment:
I think we don't need authn for tcp.  not need to add this TODO

Reply:
I can remove TODO, but I would like to align them somehow. Here, for TCP, we try to get user id from x509 directly, which seem weird.